### DISCUSSION TITLE
handling PDF reponse

### DISCUSSION THREAD

**Discussion by plutext** [Jinntec/Fore#79] [Q&A]
My use case is to submit some data and get a PDF back from the server, which I'd like the user to see.

What's the most elegant way to handle this?
 
`replace="all"` results in a blank page which says only "[object Blob]".

`replace=redirect` I haven't tried, but I guess that might be a workaround

**Comment by plutext** [comment]
`replace=redirect` didn't seem to work either.  

I doubt this is helpful, but:
```
submit-done { target: null, isTrusted: false, detail: {}, eventPhase: 0, bubbles: true, cancelable: false, returnValue: true, defaultPrevented: false, composed: true, timeStamp: 3062, … }
ForeElementMixin.js:253:13
The connection to http://localhost:8090/__es-dev-server__/message-channel was interrupted while the page was loading. join.html:230:24
```

**Reply by JoernT** [reply]
Hi @plutext 

`replace="all"` won't do it for you as it expects the response to be html.

Here is how it's currently implemented:
```
    if (this.replace === 'all') {
      document.getElementsByTagName('html')[0].innerHTML = data;
    }
```

This is still a bit a 'naive' implementation and some content-type checking on the response would certainly good. Due to time constraints i have to make some cases work first and worry about details later. (we accepting donations soon ;)

`replace="redirect"` would be the choice for you but you have to return the absolute URL to the pdf you want to display. Then it should actually work.

These are the relevant lines in the source of `fx-submission`:
```
    if (this.replace === 'redirect') {
      window.location.href = data;
    }
```

`data` would be a simple string (the URL) and should be returned from the server as response with content-type="text/plain".
